src/Sharpliner/AzureDevOps/Model/Step.cs

```diff
@@ -54,13 +54,28 @@ public abstract record Step
     /// </summary>
     [YamlIgnore]
     [DisallowNull]
-    public Conditioned<TimeSpan>? Timeout { get; init; }
+    public Conditioned<TimeSpan>? Timeout
+    {
+        get;
+        init
+        {
+            field = value;
+            if (value?.Definition is not null)
+            {
+                TimeoutInMinutes = (int)value.Definition.TotalMinutes;
+            }
+            else if (value?.Condition is not null)
+            {
+                TimeoutInMinutes = new Conditioned<int>(default, value.Condition);
+            }
+        }
+    }
 
     /// <summary>
     /// Timeout after which the step will be stopped.
     /// </summary>
     [YamlMember(Order = 210)]
-    public Conditioned<int>? TimeoutInMinutes => Timeout?.Definition != null ? (int)Timeout.Definition.TotalMinutes : null;
+    public Conditioned<int>? TimeoutInMinutes { get; init; }
```

Comment:
Should this be reverted then?

Reply:
I think this needs to stay as the `TimeoutInMinutes` property delegates to this property so here we save the actual value

Reply:
Can it be private init only?

Reply:
by having both properties have the `init` keyword we allow setting value to an integer for minutes or using a TimeSpan, the `get` here is used like any other property.

do you mean something like this?
```cs
public Conditioned<int>? TimeoutInMinutes { private get; init; }
```

Reply:
Ah, if we want both to be settable then why not, I just thought one would be settable for the user only